aya/src/sys/mod.rs

```diff
@@ -86,11 +86,12 @@ impl std::fmt::Debug for Syscall<'_> {
     }
 }
 
-fn syscall(call: Syscall<'_>) -> SysResult<c_long> {
+fn syscall(call: Syscall<'_>) -> SysResult<i64> {
     #[cfg(test)]
     return TEST_SYSCALL.with(|test_impl| unsafe { test_impl.borrow()(call) });
 
     #[cfg_attr(test, allow(unreachable_code))]
+    #[allow(clippy::useless_conversion)]
```

Comment:
where is the useless conversion? this probably can be moved to a better place?

Reply:
It's the `.into()` at the bottom. I tried assigning a local but then we need more `allow(unreachable)`s.

Reply:
Alright I've narrowed this.